sdks/python/apache_beam/runners/portability/stager_test.py

```diff
@@ -420,134 +375,167 @@ def test_sdk_location_local_directory_not_present(self):
+    self.create_temp_file(os.path.join(source_dir, 'whl.whl'), 'nothing')
     self.create_temp_file(
-        os.path.join(source_dir, 'abc.tar.gz'), 'nothing')
-    self.create_temp_file(
-        os.path.join(source_dir, 'xyz.tar.gz'), 'nothing')
-    self.create_temp_file(
-        os.path.join(source_dir, 'xyz2.tar'), 'nothing')
-    self.create_temp_file(
-        os.path.join(source_dir, 'whl.whl'), 'nothing')
-    self.create_temp_file(
-        os.path.join(source_dir, dependency.EXTRA_PACKAGES_FILE), 'nothing')
+        os.path.join(source_dir, stager.EXTRA_PACKAGES_FILE), 'nothing')
 
     options = PipelineOptions()
-    options.view_as(GoogleCloudOptions).staging_location = staging_dir
     self.update_options(options)
     options.view_as(SetupOptions).extra_packages = [
         os.path.join(source_dir, 'abc.tar.gz'),
         os.path.join(source_dir, 'xyz.tar.gz'),
         os.path.join(source_dir, 'xyz2.tar'),
-        os.path.join(source_dir, 'whl.whl'),
-        'gs://my-gcs-bucket/gcs.tar.gz']
+        os.path.join(source_dir, 'whl.whl'), '/tmp/remote/remote_file.tar.gz'
+    ]
 
-    gcs_copied_files = []
+    remote_copied_files = []
 
-    def file_copy(from_path, to_path):
-      if from_path.startswith('gs://'):
-        gcs_copied_files.append(from_path)
+    # We can not rely on actual remote file systems paths hence making
+    # '/tmp/remote/' a new remote path.
+    def is_remote_path(dummy_self, path):
+      return path.startswith('/tmp/remote/')
+
+    def file_copy(dummy_self, from_path, to_path):
+      if is_remote_path(dummy_self, from_path):
+        remote_copied_files.append(from_path)
         _, from_name = os.path.split(from_path)
         if os.path.isdir(to_path):
           to_path = os.path.join(to_path, from_name)
         self.create_temp_file(to_path, 'nothing')
-        logging.info('Fake copied GCS file: %s to %s', from_path, to_path)
-      elif to_path.startswith('gs://'):
-        logging.info('Faking file_copy(%s, %s)', from_path, to_path)
+        logging.info('Fake copied remote file: %s to %s', from_path, to_path)
+      elif is_remote_path(dummy_self, to_path):
+        logging.info('Faking upload_file(%s, %s)', from_path, to_path)
       else:
         shutil.copyfile(from_path, to_path)
 
-    dependency._dependency_file_copy = file_copy
-
-    self.assertEqual(
-        ['abc.tar.gz', 'xyz.tar.gz', 'xyz2.tar', 'whl.whl', 'gcs.tar.gz',
-         dependency.EXTRA_PACKAGES_FILE],
-        dependency.stage_job_resources(options))
-    with open(os.path.join(staging_dir, dependency.EXTRA_PACKAGES_FILE)) as f:
-      self.assertEqual(['abc.tar.gz\n', 'xyz.tar.gz\n', 'xyz2.tar\n',
-                        'whl.whl\n', 'gcs.tar.gz\n'], f.readlines())
-    self.assertEqual(['gs://my-gcs-bucket/gcs.tar.gz'], gcs_copied_files)
+    with mock.patch(
+        '.'.join([
+            self.__module__, TestStager.__name__,
+            TestStager.stage_artifact.__name__
+        ]), file_copy):
```

Comment:
Can we use a fake implementation of stage_artifact that we implemented in TestStager? Why do we have to mock it here?

Reply:
removed.